### ISSUE TITLE
HTTPError: 401 Client Error: UNAUTHORIZED for url: https://ftx.com/api/wallet/balances

### ISSUE THREAD

**Issue by Mike1999aa** [ccxt/ccxt#9537]
MUST READ THIS BEFORE SUBMITTING ISSUES (read the links, then delete this message before submitting):

- https://github.com/ccxt/ccxt/wiki/FAQ
- https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#how-to-submit-an-issue

Make sure your local version of CCXT is up to date. Check by comparing the output of `ccxt.version` to https://github.com/ccxt/ccxt/blob/master/package.json#L3

- OS: Window 10
- Programming Language version: Python 
- CCXT version: 1.52.16

```
exchange = ccxt.ftx({
    'apiKey' : apiKey ,
    'secret' : secret ,
    'enableRateLimit': True ,
    'hostname': 'ftx.com'
})
if subaccount == "0":
    print("This is Main Account")
else:
    exchange.headers = {
   'FTX-SUBACCOUNT': subaccount,
  }

exchange.verbose = True
exchange.fetch_balance()
```

```
Request: GET https://ftx.com/api/wallet/balances {'FTX-KEY': 'XXXXXX', 'FTX-TS': '1625816463042', 'FTX-SIGN': 'f4f38e578a8ade9fb0b8dbfa173f6bf33459f03e40e900feab5f8ba437c02ef1', 'FTX-SUBACCOUNT': 'XXXX', 'User-Agent': 'python-requests/2.25.1', 'Accept-Encoding': 'gzip, deflate'} None

Response: GET https://ftx.com/api/wallet/balances 401 {'Date': 'Fri, 09 Jul 2021 07:41:03 GMT', 'Content-Type': 'application/json', 'Content-Length': '42', 'Connection': 'keep-alive', 'Cache-Control': 'private, max-age=0, no-cache', 'account-id': 'None', 'Vary': 'Origin, Accept-Encoding', 'CF-Cache-Status': 'MISS', 'Expect-CT': 'max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"', 'Strict-Transport-Security': 'max-age=31536000; includeSubDomains; preload', 'X-Content-Type-Options': 'nosniff', 'Server': 'cloudflare', 'CF-RAY': '66bfe85e0f4d55fb-SIN', 'alt-svc': 'h3-27=":443"; ma=86400, h3-28=":443"; ma=86400, h3-29=":443"; ma=86400, h3=":443"; ma=86400'} {"success":false,"error":"Not logged in"}

HTTPError                                 Traceback (most recent call last)
c:\users\msif.rs.intern.3\python\python36\lib\site-packages\ccxt\base\exchange.py in fetch(self, url, method, headers, body)
    592             self.logger.debug("%s %s, Response: %s %s %s", method, url, http_status_code, headers, http_response)
--> 593             response.raise_for_status()
    594 

c:\users\msif.rs.intern.3\python\python36\lib\site-packages\requests\models.py in raise_for_status(self)
    942         if http_error_msg:
--> 943             raise HTTPError(http_error_msg, response=self)
    944 

HTTPError: 401 Client Error: UNAUTHORIZED for url: https://ftx.com/api/wallet/balances

During handling of the above exception, another exception occurred:

AuthenticationError                       Traceback (most recent call last)
<ipython-input-57-00174a29ff76> in <module>
----> 1 exchange.fetch_balance()

c:\users\msif.rs.intern.3\python\python36\lib\site-packages\ccxt\ftx.py in fetch_balance(self, params)
    900     def fetch_balance(self, params={}):
    901         self.load_markets()
--> 902         response = self.privateGetWalletBalances(params)
    903         #
    904         #     {

c:\users\msif.rs.intern.3\python\python36\lib\site-packages\ccxt\base\exchange.py in inner(_self, params)
    459                             if params is not None:
    460                                 inner_kwargs['params'] = params
--> 461                             return entry(_self, **inner_kwargs)
    462                         return inner
    463                     to_bind = partialer()

c:\users\msif.rs.intern.3\python\python36\lib\site-packages\ccxt\base\exchange.py in request(self, path, api, method, params, headers, body)
    484     def request(self, path, api='public', method='GET', params={}, headers=None, body=None):
    485         """Exchange.request is the entry point for all generated methods"""
--> 486         return self.fetch2(path, api, method, params, headers, body)
    487 
    488     @staticmethod

c:\users\msif.rs.intern.3\python\python36\lib\site-packages\ccxt\base\exchange.py in fetch2(self, path, api, method, params, headers, body)
    480         self.lastRestRequestTimestamp = self.milliseconds()
    481         request = self.sign(path, api, method, params, headers, body)
--> 482         return self.fetch(request['url'], request['method'], request['headers'], request['body'])
    483 
    484     def request(self, path, api='public', method='GET', params={}, headers=None, body=None):

c:\users\msif.rs.intern.3\python\python36\lib\site-packages\ccxt\base\exchange.py in fetch(self, url, method, headers, body)
    607         except HTTPError as e:
    608             details = ' '.join([self.id, method, url])
--> 609             self.handle_errors(http_status_code, http_status_text, url, method, headers, http_response, json_response, request_headers, request_body)
    610             self.handle_http_status_code(http_status_code, http_status_text, url, method, http_response)
    611             raise ExchangeError(details) from e

c:\users\msif.rs.intern.3\python\python36\lib\site-packages\ccxt\ftx.py in handle_errors(self, code, reason, url, method, headers, body, response, requestHeaders, requestBody)
   1809             feedback = self.id + ' ' + body
   1810             error = self.safe_string(response, 'error')
-> 1811             self.throw_exactly_matched_exception(self.exceptions['exact'], error, feedback)
   1812             self.throw_broadly_matched_exception(self.exceptions['broad'], error, feedback)
   1813             raise ExchangeError(feedback)  # unknown message

c:\users\msif.rs.intern.3\python\python36\lib\site-packages\ccxt\base\exchange.py in throw_exactly_matched_exception(self, exact, string, message)
    498     def throw_exactly_matched_exception(self, exact, string, message):
    499         if string in exact:
--> 500             raise exact[string](message)
    501 
    502     def throw_broadly_matched_exception(self, broad, string, message):

AuthenticationError: ftx {"success":false,"error":"Not logged in"}
```

**Comment by kroitor** [comment]
Duplicate:

- https://github.com/ccxt/ccxt/issues/7554#issuecomment-686152859
- https://github.com/ccxt/ccxt/search?q=ftx+Not+logged+in&type=issues

TL;DR: make sure you use the proper keys (don't confuse subaccount keys with master keys), also, try with a new keypair and make sure that your system clock is in synch with the exchange. Hope that helps, let us know if not.
